### PR TITLE
Find channel

### PR DESCRIPTION
**Pull request by ColorfulCodes**
This PR should Identify channels by ID. I am not sure if this is how you all expected it. Output looks like this: Project { channel_id: 'c5678g' }. I am writing tests as well but wanted to review this first to make sure this is correct.

### CODE REVIEW

File: skills/findChannel.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  }
+
+}
+
+async function getNewProject(x){
+  let project = await Project.findByChannel(x);
+  console.log(project)
+}
+getNewProject('c5678g')
```
Comment (cromwellryan) [inline skills/findChannel.js]: Looks like you might be locally testing by running this file through node. The class you’ve defined here should be consumed as an `import`. Let’s remove this debugging code and export our class.

👏

File: skills/findChannel.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Project{
+  constructor(channel){
+    this.channel_id = channel;
+  }
+
+  static findByChannel(channel){
+    return new Promise(function(resolve,reject){
+      let p = new Project(channel)
```
Comment (cromwellryan) [inline skills/findChannel.js]: Single letter variables don't provide much for the reader. You're writing for future you and future developers who know nothing of your intent. Make it clear.

File: skills/findChannel.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    // p.channel_id = channel;
```
[the remark this answers is not in the document]
Comment (ColorfulCodes) [reply skills/findChannel.js]: Will do.

File: skills/findChannel.js (hunk 2 above)
Comment (ColorfulCodes) [reply skills/findChannel.js]: ok, will do.

File: skills/findChannel.js (hunk 1 above)
Comment (ColorfulCodes) [reply skills/findChannel.js]: Okay, will do.

File: skills/findChannel.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    return new Promise(function(resolve,reject){
+      let projectChannel = new Project(channel)
+      resolve(projectChannel);
+    });
+
+  }
+
+}
+
+async function getNewProject(channelID){
```
Comment (cromwellryan) [inline skills/findChannel.js]: Let’s remove this since it’s not part of the app

File: skills/findChannel.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class Project{
```
Comment (cromwellryan) [inline skills/findChannel.js]: I don’t think `/skills` is the best place for this. Maybe talk to Nathan and Colby about the right place for things like this.

File: test/controllers/findChannelTests.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const { expect } = require('chai');
+const Project = require('../../skills/findChannel.js');
+
+
+describe('Find Channel Tests', () => {
+  it('tests if channel exists', async () => {
+    console.log('1',Project)
```
Comment (cromwellryan) [inline test/controllers/findChannelTests.js]: We should never leave console.logs in our production code like this. If the test assertions are not clear enough, let’s resolve that

File: skills/findChannel.js (hunk 4 above)
Comment (ColorfulCodes) [reply skills/findChannel.js]: ok

File: skills/findChannel.js (hunk 5 above)
Comment (ColorfulCodes) [reply skills/findChannel.js]: ok perfect. Nathan told me to put it in a lib file

File: test/controllers/findChannelTests.js (hunk 6 above)
Comment (ColorfulCodes) [reply test/controllers/findChannelTests.js]: will do

**Comment by cromwellryan** [comment]
What is that last commit?

File: lib/findChannel.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+class Project{
```
[the remark this answers is not in the document]
Comment (ColorfulCodes) [reply lib/findChannel.js]: done

File: test/controllers/findChannelTests.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const { expect } = require('chai');
+const Project = require('../../skills/findChannel.js');
+
+
+describe('Find Channel Tests', () => {
```
[the remark this answers is not in the document]
Comment (ColorfulCodes) [reply test/controllers/findChannelTests.js]: Okay I changed the name but Nathan told me not to put it in the controllers file.

**Pull request merged**